### PR TITLE
Fixed #25349 - ModelForm bug that occurs when saving None value to a ForeignKey with null=True, blank=False and required=False.

### PR DESCRIPTION
**Pull request by haxoza**
https://code.djangoproject.com/ticket/25349

### CODE REVIEW

File: django/db/models/fields/related_descriptors.py [hunk 1]
```diff
@@ -193,14 +193,7 @@ def __set__(self, instance, value):
         - ``instance`` is the ``child`` instance
         - ``value`` in the ``parent`` instance on the right of the equal sign
         """
-        # If null=True, we can assign null here, but otherwise the value needs
-        # to be an instance of the related class.
-        if value is None and self.field.null is False:
```
Comment (timgraham) [inline django/db/models/fields/related_descriptors.py]: I don't think removing this check is the right thing to do.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: django/db/models/fields/related_descriptors.py (hunk 1 above)
Comment (haxoza) [inline django/db/models/fields/related_descriptors.py]: I described this change in a more detailed way here:

https://code.djangoproject.com/ticket/25349#comment:6

Comment (timgraham) [inline django/db/models/fields/related_descriptors.py]: Thanks for the explanation. I raised it [on the mailing list](https://groups.google.com/d/topic/django-developers/4UL_FkDXX-w/discussion) and there is consensus to remove it. I think it should be done as a separate ticket for clarity (also needs to be documented). Are you able to do it? Here's the ticket: https://code.djangoproject.com/ticket/26179

**Comment by timgraham** [comment]
Ready for a rebase after 04e13c89138d48c20e774a2b6bf06796f73ac0fe

**Comment by timgraham** [comment]
updated and merged in 375e1cfe2b2e1c3c57f882147c34902c6e8189ac, thanks!

**Pull request closed without merging**